File: knowledgebase-sdk/src/main/java/ru/usedesk/knowledgebase_sdk/internal/data/repository/IKnowledgeBaseRepository.java
package ru.usedesk.knowledgebase_sdk.internal.data.repository;

import android.support.annotation.NonNull;

import java.util.List;

import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskDataNotFoundException;
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskHttpException;
import ru.usedesk.knowledgebase_sdk.external.entity.ArticleBody;
import ru.usedesk.knowledgebase_sdk.external.entity.ArticleInfo;
import ru.usedesk.knowledgebase_sdk.external.entity.Category;
import ru.usedesk.knowledgebase_sdk.external.entity.SearchQuery;
import ru.usedesk.knowledgebase_sdk.external.entity.Section;

public interface IKnowledgeBaseRepository {
    @NonNull
    List<Section> getSections(@NonNull String id, @NonNull String token) throws UsedeskHttpException;

    @NonNull
    ArticleBody getArticleBody(@NonNull String id, @NonNull String token, long articleId) throws UsedeskHttpException;

    @NonNull
    List<ArticleBody> getArticles(@NonNull String id, @NonNull String token, @NonNull SearchQuery searchQuery) throws UsedeskHttpException;

    @NonNull
    List<Category> getCategories(@NonNull String id, @NonNull String token, long sectionId) throws UsedeskHttpException, UsedeskDataNotFoundException;

    @NonNull
    List<ArticleInfo> getArticles(@NonNull String id, @NonNull String token, long categoryId) throws UsedeskHttpException, UsedeskDataNotFoundException;

    void addViews(@NonNull String accountId, @NonNull String token, long articleId) throws UsedeskHttpException, UsedeskDataNotFoundException;
}

File: knowledgebase-sdk/src/main/java/ru/usedesk/knowledgebase_sdk/internal/data/repository/IKnowledgeBaseRepository.kt
package ru.usedesk.knowledgebase_sdk.internal.data.repository

import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskDataNotFoundException
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskHttpException
import ru.usedesk.knowledgebase_sdk.external.entity.*

interface IKnowledgeBaseRepository {
    @Throws(UsedeskHttpException::class)
    fun getSections(accountId: String, token: String): List<UsedeskSection>

    @Throws(UsedeskHttpException::class)
    fun getArticleBody(accountId: String, token: String, articleId: Long): UsedeskArticleBody

    @Throws(UsedeskHttpException::class)
    fun getArticles(accountId: String, token: String, searchQuery: UsedeskSearchQuery): List<UsedeskArticleBody>

    @Throws(UsedeskHttpException::class, UsedeskDataNotFoundException::class)
    fun getCategories(accountId: String, token: String, sectionId: Long): List<UsedeskCategory>

    @Throws(UsedeskHttpException::class, UsedeskDataNotFoundException::class)
    fun getArticles(accountId: String, token: String, categoryId: Long): List<UsedeskArticleInfo>

    @Throws(UsedeskHttpException::class, UsedeskDataNotFoundException::class)
    fun addViews(accountId: String, token: String, articleId: Long)
}